### PR TITLE
Expose options

### PR DESCRIPTION
**Pull request by martinthomson**
This one builds on #182 to also address #183 and #184.

@kitcambridge, does this address your concerns?

### CODE REVIEW

**Comment by ghost** [comment]
Yes, LGTM. Thank you!

File: index.html [hunk 1]
```diff
@@ -558,6 +570,15 @@
         with a <code><a>DOMException</a></code> whose name is "<code><a>SecurityError</a></code>"
         and terminate these steps.
         </li>
+        <li>Let <var>allOptions</var> be the value of the options argument, if provided, or a
```
Comment (beverloo) [inline index.html]: <code>options</code>

File: index.html [hunk 2]
```diff
@@ -674,26 +709,59 @@
...
           <a>user agent</a> SHOULD enforce it on incoming <a>push messages</a>.
         </p>
-        <dl title="dictionary PushSubscriptionOptions" class="idl">
+        <p>
+          Once set, options for a <a>push subscription</a> cannot change. A pre-existing <a>push
+          subscription</a> can be <a href=
+          "#widl-PushSubscription-unsubscribe-Promise-boolean">unsubscribed</a> to create a <a>push
+          subscription</a> with new options.
+        </p>
+        <dl title="dictionary PushSubscriptionOptionsDict" class="idl" data-merge=
```
Comment (beverloo) [inline index.html]: PushSubscriptionInit? PushSubscribeOptions? I'd love to find a way to get rid of the *Dict suffix.

File: index.html [hunk 3]
```diff
@@ -708,6 +776,9 @@
           readonly attribute USVString endpoint
         </dt>
         <dt>
+          [SameObject] readonly attribute PushSubscriptionOptions
```
Comment (beverloo) [inline index.html]: Attribute name? `options`?

File: index.html [hunk 4]
```diff
@@ -816,6 +887,9 @@
         <li>Return <var>map</var>.
         </li>
       </ol>
+      <p>
+        Note that the options to a <code><a>PushSubscription</a></code> are not serialized.
```
Comment (beverloo) [inline index.html:892]: What does this mean?

**Edit**: Never mind, JSON.serialize(pushSubscription).

File: index.html [hunk 5]
```diff
@@ -674,26 +709,59 @@
...
+          option includes an elliptic curve public key for an application server. This is the key
+          that the application server will use to authenticate itself when sending push messages to
+          this subscription as defined in [[!WEBPUSH-VAPID]]; the push service will reject any push
+          message unless the corresponding private key is used to generate an authentication token.
+        </p>
+        <p>
+          If present, the value of <code><a>applicationServerKey</a></code> MUST include a point on
+          the P-256 elliptic curve [[!FIPS-186-3]], encoded in the uncompressed form described in
+          [[!X9.62]] Annex A (that is, 65 octets, starting with an 0x04 octet). A different key to
+          the one used for message encryption [[WEBPUSH-ENCRYPTION]] is strongly recommended.
```
Comment (beverloo) [inline index.html]: This MUST be a different key than the one used for message encryption [[WEBPUSH-ENCRYPTION]].

**Comment by beverloo** [comment]
Thanks!

@mvano PTAL?

**Comment by mvano** [comment]
Looks good, @martinthomson could you please address @beverloo's comments? After that this is fine to merge. Thanks!

File: index.html (hunk 3 above)
Comment (martinthomson) [inline index.html]: Amazing.  You need one of those? :)

**Comment by martinthomson** [comment]
@mvano, updated, thanks.

**Pull request merged**